backend/app/repositories/validation.py

```diff
@@ -0,0 +1,113 @@
+from typing import Any, Optional
+
+from app.repositories.base import BaseRepository, get_repository
+from fastapi import HTTPException, status
+from app.models.validation import Validation
+from app.settings import settings
+
+
+class ValidationRepository(BaseRepository[Validation]):
+    model_class = Validation
+    index = settings.VALIDATION_INDEX
+
+    def query_by_filter(
+        self,
+        datasource_id: str = None,
+        dataset_id: str = None,
+        period: int = 14,
+    ):
+        query = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {"range": {"meta.run_id.run_time": {"gte": f"now-{period}d", "lte": "now"}}},
+                    ]
+                }
+            },
+            "sort": [{"meta.run_id.run_time": "asc"}]
+        }
+        if not dataset_id and not datasource_id:
```

Comment:
We shouldn't raise `HTTPException` in those methods. The database layer should be independent of the API layer. If you need to validate this, do it directly in the endpoint or in a dependency.

Reply:
Agreed, moved it out to the API layer.